packages/airnode-validator/src/ois/ois.test.ts

```diff
@@ -59,22 +59,98 @@ describe('disallows reserved parameter name', () => {
   });
 });
 
-it('verifies that the same parameter cannot be in fixedOperationParameters and parameters', () => {
-  const ois = loadOisFixture();
-  ois.endpoints[0].fixedOperationParameters.push({
-    operationParameter: ois.endpoints[0].parameters[0].operationParameter,
-    value: '123',
+describe('parameter uniqueness', () => {
+  it('allows parameter with same name, but different http method', () => {
+    const paramName = 'some-id';
+    const ois = loadOisFixture();
+    ois.apiSpecifications.paths['/convert'].get!.parameters.push({
+      in: 'query',
+      name: paramName,
+    });
+    ois.endpoints[0].fixedOperationParameters.push({
+      operationParameter: {
+        in: 'query',
+        name: paramName,
+      },
+      value: 'query-id',
+    });
+    ois.apiSpecifications.paths['/convert'].get!.parameters.push({
+      in: 'cookie',
+      name: paramName,
+    });
+    ois.endpoints[0].fixedOperationParameters.push({
+      operationParameter: {
+        in: 'cookie',
+        name: paramName,
+      },
+      value: 'cookie-id',
+    });
+
+    expect(() => oisSchema.parse(ois)).not.toThrow();
   });
 
-  expect(() => oisSchema.parse(ois)).toThrow(
-    new ZodError([
-      {
-        code: 'custom',
-        message: 'Parameters "from" are used in both "parameters" and "fixedOperationOperations"',
-        path: ['ois', 'endpoints', 0],
-      },
-    ])
-  );
+  it(`fails if the same parameter is used in "parameters"`, () => {
+    const ois = loadOisFixture();
+    ois.endpoints[0].parameters.push(ois.endpoints[0].parameters[0] as any);
+
+    expect(() => oisSchema.parse(ois)).toThrow(
+      new ZodError([
+        {
```

Comment:
I'm deliberately including both errors because it's easier for the user to find the offending pair... I'd even consider adding more details to the message (e.g. `Parameter "param-name" in "query" is used multiple times`). WDYT?

Reply:
Yeah. I've added it. Btw. feel free to review if you have time :)